Duplicate metrics collector registration attempted

```
panic: duplicate metrics collector registration attempted

goroutine 68 [running]:
github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0xc000568f70, {0xc000568fb0, 0x1, 0x0})
    github.com/prometheus/client_golang@v1.12.1/prometheus/registry.go:403 +0x7f
github.com/prometheus/client_golang/prometheus/promauto.Factory.NewCounterVec({{0x1d9a6e8, 0xc00022d590}}, {{0x0, 0x0}, {0x0, 0x0}, {0x1b77405, 0x29}, {0x1b65bd8, 0x20}, ...}, ...)
    github.com/prometheus/client_golang@v1.12.1/prometheus/promauto/auto.go:276 +0x133
github.com/parca-dev/parca-agent/pkg/profiler.NewCgroupProfiler({0x1d78d20, 0xc00022d4f0}, {0x1d9a6e8, 0xc00022d590}, 0xc000307740, {0x1d7a1e0, 0xc00023bf40}, {0x1d94610, 0xc000569bf0}, 0xc0005a1230, ...)
    github.com/parca-dev/parca-agent/pkg/profiler/profiler.go:103 +0x465
github.com/parca-dev/parca-agent/pkg/target.(*ProfilerPool).Sync(0xc000566ea0, {0xc000648000, 0x9, 0x3})
    github.com/parca-dev/parca-agent/pkg/target/profiler_pool.go:132 +0x266
github.com/parca-dev/parca-agent/pkg/target.(*Manager).reconcileTargets(0xc000648e00, {0x1dadc70, 0xc000591000}, 0x0)
    github.com/parca-dev/parca-agent/pkg/target/manager.go:97 +0x1eb
github.com/parca-dev/parca-agent/pkg/target.(*Manager).Run(0xc000308dc0, {0x1dadc70, 0xc000591000}, 0xc0002d0660)
    github.com/parca-dev/parca-agent/pkg/target/manager.go:75 +0xc9
main.main.func8()
    github.com/parca-dev/parca-agent/cmd/parca-agent/main.go:370 +0x105
github.com/oklog/run.(*Group).Run.func1({0xc000591040, 0xc0001bca20})
    github.com/oklog/run@v1.1.0/group.go:38 +0x2f
created by github.com/oklog/run.(*Group).Run
    github.com/oklog/run@v1.1.0/group.go:37 +0x22f
[event: pod parca/parca-agent-4h6gm] Back-off restarting failed container
```

cc @Sylfrena @metalmatze

--- a/pkg/profiler/profiler.go
+++ b/pkg/profiler/profiler.go
@@ -100,8 +100,9 @@ func NewCgroupProfiler(
 		mtx: &sync.RWMutex{},
 		missingStacks: promauto.With(reg).NewCounterVec(
 			prometheus.CounterOpts{
-				Name: "parca_agent_profiler_missing_stacks_total",
-				Help: "Number of missing profile stacks",
+				Name:        "parca_agent_profiler_missing_stacks_total",
+				Help:        "Number of missing profile stacks",
+				ConstLabels: map[string]string{"target": target.String()},
 			},
 			[]string{"type"}),
 	}

